src/selectors/ProfilePageSelector.js

```diff
@@ -97,24 +83,231 @@ export const editableFormSelector = createSelector(
+      });
+    }
+    return linksByPlatform;
+  },
+);
+
+// Fleshes out our list of existing social links with all the other ones the user can set.
+export const formSocialLinksSelector = createSelector(
+  socialLinksByPlatformSelector,
+  draftSocialLinksByPlatformSelector,
+  (linksByPlatform, draftLinksByPlatform) => {
+    const knownPlatforms = ['twitter', 'facebook', 'linkedin'];
+    const socialLinks = [];
+    // For each known platform
+    knownPlatforms.forEach((platform) => {
+      // If the link is in our drafts.
+      if (draftLinksByPlatform[platform] !== undefined) {
+        // Use the draft one.
+        socialLinks.push(draftLinksByPlatform[platform]);
+      } else if (linksByPlatform[platform] !== undefined) {
+        // Otherwise use the real one.
+        socialLinks.push(linksByPlatform[platform]);
+      } else {
+        // And if it's not in either, use a stub.
+        socialLinks.push({
+          platform,
+          socialLink: null,
+        });
+      }
+    });
+    return socialLinks;
+  },
+);
+
+export const visibilitiesSelector = createSelector(
+  profilePreferencesSelector,
+  accountPrivacySelector,
+  (preferences, accountPrivacy) => {
+    switch (accountPrivacy) {
+      case 'all_users':
+        return {
+          visibilityBio: 'all_users',
+          visibilityCourseCertificates: 'all_users',
+          visibilityCountry: 'all_users',
+          visibilityEducation: 'all_users',
+          visibilityLanguageProficiencies: 'all_users',
+          visibilityName: 'all_users',
+          visibilitySocialLinks: 'all_users',
+        };
+      case 'custom':
+        return {
+          visibilityBio: preferences.visibilityBio || 'private',
+          visibilityCourseCertificates: preferences.visibilityCourseCertificates || 'private',
+          visibilityCountry: preferences.visibilityCountry || 'private',
+          visibilityEducation: preferences.visibilityEducation || 'private',
+          visibilityLanguageProficiencies: preferences.visibilityLanguageProficiencies || 'private',
+          visibilityName: preferences.visibilityName || 'private',
+          visibilitySocialLinks: preferences.visibilitySocialLinks || 'private',
+        };
+      case 'private':
+      default:
+        // If there is some other value for accountPrivacy set, we're going to assume
+        // it's private, since that's safest.
+        return {
+          visibilityBio: 'private',
+          visibilityCourseCertificates: 'private',
+          visibilityCountry: 'private',
+          visibilityEducation: 'private',
+          visibilityLanguageProficiencies: 'private',
+          visibilityName: 'private',
+          visibilitySocialLinks: 'private',
+        };
+    }
+  },
+);
+
+/**
+ * If there's no draft present at all (undefined), use the original committed value.
+ */
+function chooseFormValue(draft, committed) {
+  return draft !== undefined ? draft : committed;
```

Comment:
`typeof draft !== 'undefined'`?

Reply:
Not really necessary to my knowledge... if the value of draft is `undefined` then its type will be `'undefined`' they're essentially the same, but this one's a bit shorter. :)